pkg/kubecost/allocation.go

```diff
@@ -151,6 +174,23 @@ func (pv *PVAllocations) Add(that PVAllocations) PVAllocations {
 	return apv
 }
 
+// Equal returns true if the two PVAllocations are equal in length and contain the same keys
+// and values.
+func (aPVs *PVAllocations) Equal(thatPVs PVAllocations) bool {
+	if len(*aPVs) != len(thatPVs) {
+		return false
+	}
```

Comment:
Should this `nil` check `aPVs`?

Reply:
Yeah, I think so! Good suggestion.

Reply:
> Neat! Why is that? Does it just take field order into account?

Yeah, basically, the fields have no names associated with the data in the format itself. It just deterministically parses the fields in order and pulls the field name from the AST data. 

> is that expected to be fine because we always clone out of Stores?
Yes, and the only real danger for races/corruption would've happened during iteration (via `Each`), and we didn't lock at all in `Each`, which I think is a fairly good hint at existing safety.